service-map/src/main/java/fi/nls/oskari/map/layer/OskariLayerServiceMybatisImpl.java

```diff
@@ -303,6 +310,23 @@ public OskariLayer find(int id) {
         }
         return Collections.emptyList();
     }
+    
+    @Override
+	public List<OskariLayer> findByDataProviderId(int dataProviderId) {
+    	LOG.debug("Find by data provider id: " + dataProviderId);
+        final SqlSession session = factory.openSession();
+        try {
```

Comment:
You could do try-with-resources and get rid of the finally statement:
```
try (SqlSession session = factory.openSession()) { ... } 
```

Reply:
Fixed